### PR TITLE
[ENH] Word List: New widget

### LINKED ISSUE #630: New widget: Word list
**Input**: Words (Data Table)
**Outputs**: Words (Table), Selected Words (Table)

Input has to include at least one string variable, from where the widget extracts the list of words (one row each, it restricts the terms to the first three words, and provides a warning about truncation). Words from the input are automatically added to the list of words, according to the Update rule (Mute means nothing happens).

Words in the list can be edited (double click), added (with plus button at the bottom), deleted (delete button or - button at the bottom). See edit domain widget for +/- buttons.

Wordlist can be saved in the library, read from the text document (More > Import Words from File...), or saved (More > Save Words to File...). Input files can be anything that can store textual content (file with words as rows?), including an ontology (@VesnaT I can help you with ontologies). (See Python Script for similar functionality.)
When the user clicks on the update button list saves to widget settings, when the user uses the Save words to file option word list saves to File (and settings do not update).

On the output, the data Data contains a Data Table with a list of Words (current word list). The output table has one column (with name words). Selected Words output has one “Words” column, and only selected words (Selected Words). Words column is a String variable and has the attribute `{“type”: “words”}` in both outputs.

pasted image 0

### PR DESCRIPTION
**Pull request by VesnaT**
##### Issue

Fixes #630

##### TODO
- [ ] Icon

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: orangecontrib/text/widgets/owwordlist.py [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+                    indices.add(int(name[len(default_name):]))
+                except ValueError:
+                    pass
+        index = min(set(range(max(indices) + 1 + 1)) - indices)
+        return f"{default_name} {index}"
+
+
+class UpdateRules:
+    INTERSECT, UNION, INPUT, LIBRARY = range(4)
+    ITEMS = ["Intersection", "Union", "Only Input", "Ignore Input"]
```
Comment (ajdapretnar) [inline orangecontrib/text/widgets/owwordlist.py:82]: I would have this as "Only input" and "Ignore input". I believe this is how we have it elsewhere. :)

File: orangecontrib/text/widgets/owwordlist.py [hunk 2]
```diff
@@ -0,0 +1,605 @@
...
+
+    class Inputs:
+        words = Input("Words", Table)
+
+    class Outputs:
+        selected_words = Output("Selected Words", Table)
+        words = Output("Words", Table)
+
+    class Warning(OWWidget.Warning):
+        no_string_vars = Msg("Input 'Words' needs at least one Text variable.")
```
Comment (ajdapretnar) [inline orangecontrib/text/widgets/owwordlist.py:160]: I would omit "Words" as this is the only input there is. Unless we plan to add more than one input?

**Comment by ajdapretnar** [comment]
Other than a few minor details (see comments) this widget works a-ok. 👌

File: orangecontrib/text/widgets/owwordlist.py (hunk 1 above)
Comment (VesnaT) [reply orangecontrib/text/widgets/owwordlist.py]: Fixed

File: orangecontrib/text/widgets/owwordlist.py (hunk 2 above)
Comment (VesnaT) [reply orangecontrib/text/widgets/owwordlist.py]: Fixed

**Pull request merged**